### DISCUSSION TITLE
Multiple errors during ArgoCD login when exposed as LoadBalancer

### DISCUSSION THREAD

**Discussion by meghanabsrinath** [argoproj/argo-cd#5411] [Q&A] [answered]
We are seeing few random, intermittent issues with the latest installation of ArgoCD on EKS/AKS/GKE clusters when argocd-server is exposed as `LoadBalancer` service. While doing argocd login, we are facing the error as below
```
WARNING: server is not configured with TLS. Proceed (y/n)? y
FATA[0045] rpc error: code = Unavailable desc = transport is closing
```
However, this works after a while without any issues. This happens even when using` --insecure` flag. Sometimes, the error is
```
FATA[0040] dial tcp 104.xx.xx.xx:443: operation was canceled
```
And again, after a while, it works perfectly. Furthermore, sometimes if login is successful with no errors and we try to update a password for any account, we get the error
```
FATA[0002] rpc error: code = Unauthenticated desc = Invalid username or password
```
But when we try to login after the error, password will actually be updated. We came across a similar issue in [#2211](https://github.com/argoproj/argo-cd/issues/2211)
ArgoCD version:
```
argocd: v1.8.3+0f9c684
  BuildDate: 2021-01-21T22:19:20Z
  GitCommit: 0f9c68427882bf4633d395cbfcd7c9271795fd9b
  GitTreeState: clean
  GoVersion: go1.14.12
  Compiler: gc
  Platform: linux/amd64
```  
kubectl version
```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.1", GitCommit:"206bcadf021e76c27513500ca24182692aabd17e", GitTreeState:"clean", BuildDate:"2020-09-09T11:26:42Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.15", GitCommit:"2a313bf59aa0e1c98bbee2aea6ae727d3f6da6fd", GitTreeState:"clean", BuildDate:"2020-09-03T05:20:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by meghanabsrinath** [comment, marked as answer]
This was an issue with the load balancer being provisioned in the cluster. The LB was not yet in Healthy state when we tried to login to the ArgoCD server and hence the errors. This is resolved now. We added a logic to check health of the LB before doing an `argocd login`.

**Comment by akhilreddynalla** [comment]
[ssm-user@ip-192-456-124-145 ~]$ argocd login 34.211.102.83:8080
FATA[0040] dial tcp 34.211.102.83:8080: operation was canceled

patch in nodeport getting same error

**Comment by shiveshabhishek** [comment]
For me, the service was nodeport and I directly hit the servicename without any ports and I was able to login

```
argocd@argocd-server-64957744c9-xpn86:~$ argocd login --username admin --password <PASSWORD> argocd-server --loglevel debug

WARNING: server certificate had error: x509: certificate signed by unknown authority. Proceed insecurely
(y/m)z y

‘admin:login' logged in successfully

Context 'argocd-server' updated
```
